src/db/connection.rs

```diff
@@ -1,15 +1,39 @@
+use crate::db::models::Permission;
+use crate::db::models::User;
 use crate::db::sql_types::*;
+use crate::types::{Token, TokenError};
+use crate::utils::*;
 use diesel::pg::PgConnection;
 use rocket::fairing::{AdHoc, Fairing};
 use rocket::logger::error;
 use rocket_contrib::databases::r2d2::{Pool, PooledConnection};
 use rocket_contrib::databases::{database_config, Poolable};
 
+pub enum Authorization {
+    Authorized {
+        user: User,
+        permissions: Vec<Permission>,
+    },
+    UnAuthorized {
```

Comment:
Eh, not in love with this, how about we do `Valid` and `Invalid`

Reply:
Done!